GVFS/GVFS.Common/ProductUpgraderPlatformStrategy.cs

```diff
@@ -0,0 +1,25 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public abstract class ProductUpgraderPlatformStrategy
+    {
+        public ProductUpgraderPlatformStrategy(PhysicalFileSystem fileSystem, ITracer tracer)
+        {
+            this.FileSystem = fileSystem;
+            this.Tracer = tracer;
+        }
+
+        protected PhysicalFileSystem FileSystem { get; private set; }

```

Comment:
nit: drop the `private set;` and it will become read-only (only assignable in the constructor).